AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs

```diff
@@ -68,6 +68,62 @@ public ClaimantInformation GetClaimantById(int claimId, int personRef)
                 : MapPersonAndAddressesToClaimantInformation(databaseRecord.person, databaseRecord.address);
         }
 
+        public List<TaxPayerInformation> GetAllTaxPayers(string firstname = null, string lastname = null, string postcode = null, string address = null)
+        {
+            var firstNameSearchPattern = GetSearchPattern(firstname);
+            var lastNameSearchPattern = GetSearchPattern(lastname);
+            var addressSearchPattern = GetSearchPattern(address);
+            var postcodeSearchPattern = GetSearchPattern(postcode);
+
+            var taxPayers = (
+                from taxPayer in _academyContext.TaxPayers
+                join occupation in _academyContext.Occupations on taxPayer.AccountRef equals occupation.AccountRef
+                join property in _academyContext.CouncilProperties on occupation.PropertyRef equals property.PropertyRef
+                join phoneNumber in _academyContext.PhoneNumbers on taxPayer.AccountRef.ToString() equals phoneNumber.Reference
+                where string.IsNullOrEmpty(firstname) || EF.Functions.ILike(taxPayer.FirstName, firstNameSearchPattern)
+                where string.IsNullOrEmpty(lastname) || EF.Functions.ILike(taxPayer.LastName, lastNameSearchPattern)
+                where string.IsNullOrEmpty(address) || EF.Functions.ILike(property.AddressLine1.Replace(" ", ""), addressSearchPattern)
+                where string.IsNullOrEmpty(postcode) || EF.Functions.ILike(property.PostCode.Replace(" ", ""), postcodeSearchPattern)
+                select new TaxPayerInformation
+                {
+                    AccountRef = taxPayer.AccountRef,
+                    Uprn = property.Uprn,
+                    FirstName = taxPayer.FirstName,
+                    LastName = taxPayer.LastName,
+                    TaxPayerAddress = new Address
+                    {
+                        AddressLine1 = property.AddressLine1,
+                        AddressLine2 = property.AddressLine2,
+                        AddressLine3 = property.AddressLine3,
+                        AddressLine4 = property.AddressLine4,
+                        PostCode = property.PostCode
+                    }.ToDomain(),
```

Comment:
This might be quicker to write this out in domain form initially, instead of writing out as an entity and then converting it using `.ToDomain()`.